### PR TITLE
bump to version 5.0.6

### PR DESCRIPTION
**Pull request by croth1**
:tada:

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ build:
 
 requirements:
   build:
-    - gcc  # [linux]
+    - toolchain
```
Comment (isuruf) [inline recipe/meta.yaml]: Can you try changing this to `toolchain3`? It is an experimental package that will set rpath for you automatically.

**Comment by croth1** [comment]
Linux dies with the following error message. Looks like the makefile forgot to add `-liconv` to the linker.

```
g++  -std=c++11 -fPIC   -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib  -o gnuplot alloc.o axis.o breaders.o boundary.o color.o command.o contour.o datablock.o datafile.o dynarray.o eval.o external.o fit.o gadgets.o getcolor.o graph3d.o graphics.o help.o hidden3d.o history.o internal.o interpol.o libcerf.o matrix.o misc.o mouse.o multiplot.o parse.o plot.o plot2d.o plot3d.o pm3d.o readline.o save.o scanner.o set.o show.o specfun.o standard.o stats.o stdfn.o tables.o tabulate.o term.o time.o unset.o util.o util3d.o variable.o version.o  wxterminal/gp_cairo.o wxterminal/gp_cairo_helpers.o  qtterminal/qt_term.o -lreadline   -lz -lgd -lgd -lm -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib -lz -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib -lpng16 -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib -lfreetype -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib -lfontconfig -lfreetype -ljpeg -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib -ltiff -lwebp   -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib -lQt5Network -lQt5Svg -lQt5PrintSupport -lQt5Widgets -lQt5Gui -lQt5Core -ldl -lm  -L/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib -lz -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lcairo -lglib-2.0
/opt/rh/devtoolset-2/root/usr/libexec/gcc/x86_64-redhat-linux/4.8.2/ld: set.o: undefined reference to symbol 'libiconv_open'
/opt/rh/devtoolset-2/root/usr/libexec/gcc/x86_64-redhat-linux/4.8.2/ld: note: 'libiconv_open' is defined in DSO /feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/libiconv.so.2 so try adding it to the linker command line
/feedstock_root/build_artefacts/gnuplot_1501358128924/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/libiconv.so.2: could not read symbols: Invalid operation
collect2: error: ld returned 1 exit status
```

**Comment by croth1** [comment]
The mac build fails is probably related to this thread:

https://sourceforge.net/p/gnuplot/mailman/message/35674574/

>It looks to me that the compiler is mistakenly resolving
#include "term.h"
to 
 /home/user/spack/opt/spack/linux-ubuntu16-x86_64/gcc-5.4.0/ncurses-6.0-edxtvp6xypxklibesnv2z5gstx3rat7a/include/term.h
rather than to
 <build-dir>/src/term.h
That is, it is pulling in a file "term.h" from the ncurses package when it should
be pulling in a file with the same name from the gnuplot source directory.

**Comment by isuruf** [comment]
> The mac build fails is probably related to this thread:

If that is the case, toolchain3 will not work as it will add the `-I${PREFIX}/include` to the front.

**Comment by jakirkham** [comment]
Could try patching `gnuplot` to rename the header and usages thereof. Though I guess that is a bit tricky as they need `ncurses` too it seems. 😕 Maybe we can ask if they would be ok renaming that header and patch accordingly?

**Comment by isuruf** [comment]
Spack uses this patch, https://github.com/LLNL/spack/pull/3185/files#diff-bb66a512f46f92e707521a32bbbe0d7d

**Comment by jakirkham** [comment]
That patch looks easy.

Also at the end of the mailing list thread, they mention this is only needed for the docs. Could possibly disable building the docs.

So there are a few options to consider.

**Comment by croth1** [comment]
thanks for your help, much greener now 👍

**Pull request merged**